Library/Homebrew/install.rb

```diff
@@ -247,6 +249,18 @@ def install_formula(
       f.print_tap_action
       build_options = f.build
 
+      if !Homebrew::EnvConfig.no_install_upgrade? && f.outdated? && !f.head?
+        outdated_formulae = [f, *f.old_installed_formulae]
+        version_upgrade = "#{f.linked_version} -> #{f.pkg_version}"
```

Comment:
Can any of this logic be shared with `upgrade.rb`? It feels like having a consistent function that's used for both places would be desirable. Thoughts?

Reply:
Agreed. Where should such a function go?

Reply:
In `upgrade.rb` I think.